tiled/server/router.py

```diff
@@ -1077,11 +1077,53 @@ async def post_metadata(
     settings: BaseSettings = Depends(get_settings),
     entry=SecureEntry(scopes=["write:metadata", "create"]),
 ):
+    for data_source in body.data_sources:
+        if data_source.assets:
+            raise HTTPException(
+                "Externally-managed assets cannot be registered "
+                "using POST /metadata/{path} Use POST /register/{path} instead."
+            )
     if not getattr(entry, "writable", False):
         raise HTTPException(
             status_code=405, detail=f"Data cannot be written at the path {path}"
         )
+    return await _create_node(
```

Comment:
Is `writable` actually needed here?  My understanding is that this route simply modifies the catalog, not the storage (file system).

```python3
    if not getattr(entry, "writable", False):
        raise HTTPException(
            status_code=405, detail=f"Data cannot be written at the path {path}"
        )
```

Reply:
I ask because it is causing the "test_container_fields.py" tests to fail for the top-level container node (with no data sources). Whereas removing this check enables the entire Tiled test suite to pass.

For comparison, the new /register route does not include this check.

Reply:
Perhaps this check is only needed if data sources are present?

```python3
    if data_sources and not getattr(entry, "writable", False):
        raise HTTPException(
            status_code=405, detail=f"Data cannot be written at the path {path}"
        )
```

Reply:
I now see that this check was to ensure that internally managed assets get the backing storage initialized. Have submitted a short [PR to this branch](https://github.com/danielballan/tiled/pull/1) using the approach in my comment above.